src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -803,8 +803,10 @@ static int rbd_nbd(int argc, const char *argv[])
       cmd = Connect;
     } else if (strcmp(*args.begin(), "unmap") == 0) {
       cmd = Disconnect;
+      g_ceph_context->_conf->set_val_or_die("admin_socket", "");
```

Comment:
@dillaman  for example, if i call:
rbd-nbd --admin-socket=/temp/test.sock map rbd/image1, 
there will have three elems in args:
1\) --admin-socket=/temp/test.sock
2\) map
3\) rbd/image1

After global_init, --admin-socket=/temp/test.sock(elem1, also, args.begin, will be removed from args), and only map and rbd/image1 are left. In line 802, it compare args.begin() to check cmd type. So If we move global_init after processing the command line, some thing will be different ...

Please correct me.